ra-optimisation/rao-commons/src/main/java/com/farao_community/farao/rao_commons/linear_optimisation/IteratingLinearOptimizer.java

```diff
@@ -147,8 +159,14 @@ private IteratingLinearOptimizerResult createResult(FlowResult flowResult,
     }
 
     private RangeActionResult roundResult(RangeActionResult rangeActionResult, Network network, IteratingLinearOptimizerResult previousResult) {
+        Map<RangeAction, Double> roundedSetPoints = new HashMap<>(rangeActionResult.getOptimizedSetPoints());
+        // Round set points to the closest integer for non-pst range actions
+        for (RangeAction rangeAction : rangeActionResult.getOptimizedSetPoints().keySet().stream().filter(ra -> !(ra instanceof PstRangeAction)).collect(Collectors.toSet())) {
+            roundedSetPoints.put(rangeAction, (double) Math.round(rangeActionResult.getOptimizedSetPoint(rangeAction)));
+        }
```

Comment:
improve readability, maybe by using two stream calls to build the new map (once on PstRangeActions, once on other RangeActions) ?